### DISCUSSION TITLE
CLAP host support

### DISCUSSION THREAD

**Discussion by alex-tee** [free-audio/clap-juce-extensions#152] [General]
I'm just wondering if this is on the menu at all? The README says it's not implemented at the moment but is it planned?

**Comment by baconpaul** [comment]
We have a branch with it sort of started but neither @jatinchowdhury18 nor i have worked on it in a while. The clap-wrapper project probably changes how I would implement it too.

So "we've thought about it" but "we haven't done it" is the short version.

Curious your use case (other than "host a plugin in juce" obviously! I meant a bit more specific than that :) )

**Comment by jatinchowdhury18** [comment]
If you're curious, this repository shows most of the progress that I had made: https://github.com/jatinchowdhury18/juce_clap_hosting.

However, as Paul mentioned, that code was written before the clap-wrapper code, so there's probably a lot that could be simplified and/or re-used from that codebase.

**Comment by baconpaul** [comment]
Maybe I should try and merge the clap wrapper bases with branch and a tweak or two I made to it and see if I can make bespoke load claps or some such

**Comment by alex-tee** [comment]
I see, thanks for the info!

>Curious your use case

@baconpaul It's for [Zrythm](https://www.zrythm.org/en/index.html). I'm considering hosting plugins via JUCE.

**Comment by baconpaul** [comment]
Gotcha. So hosting claps directly is really pretty straight forward and will give you higher fidelity than hosting via juce (you'll get access to features like polyphonic modulation and sample accurate parameters which the juce hosting sort of will have to blur away or will expose as native CLAP like we do on plugin side). 

For protocols like AU and VST3 I get not wanting to grapple with them as much (although AU hosting is pretty easy to host if you just do it the mac os way). For CLAP in ZRythhm I wonder if a direct hosting mode wouldnt' just be easier?

I know you already host through some mechanism (I can see a familiar plugin on your home page :) ) I guess right now you are using Carla right?

**Reply by alex-tee** [reply]
I think you're right, hosting CLAP directly might be a better idea, but it's low on my priority list so I'd use a JUCE module for the time being if it existed.

Yeah, I'm using Carla currently. Thanks for your work on Surge btw!
